### PR TITLE
a few improvements for ROCm support

### PR DESCRIPTION
**Pull request by nevion**

### CODE REVIEW

**Comment by pavanky** [comment]
build arrayfire ci

**Comment by pavanky** [comment]
build arrayfire windows ci

File: src/backend/opencl/kernel/bilateral.hpp [hunk 1]
```diff
@@ -49,7 +49,8 @@ void bilateral(Param out, const Param in, float s_sigma, float c_sigma)
 
     std::call_once( compileFlags[device], [device] () {
             bool use_native_exp = (getActivePlatform() != AFCL_PLATFORM_POCL
-                                && getActivePlatform() != AFCL_PLATFORM_APPLE);
+                                && getActivePlatform() != AFCL_PLATFORM_APPLE
+                                && getActivePlatform() != AFCL_PLATFORM_AMD);
```
Comment (pavanky) [inline src/backend/opencl/kernel/bilateral.hpp]: This is definitely supported in older versions. I think a better check is needed here.

**Comment by nevion** [comment]
@pavanky  No doubt supported in the past against their dead backend (AMD APPSDK).  Not sure about amdgpu-pro but anyway it's either that or another product that is soley for AMD products that also passes the existing AMD platform mapping.  I added platform support for that and then decided it was better not to.  Besides, why encourage a fast crap accuracy exp over a fast high accuracy exp.

File: src/backend/opencl/kernel/bilateral.hpp (hunk 1 above)
Comment (nevion) [inline src/backend/opencl/kernel/bilateral.hpp]: @pavanky No doubt supported in the past against their dead backend (AMD APPSDK). Not sure about amdgpu-pro but anyway it's either that or another product that is soley for AMD products that also passes the existing AMD platform mapping. I added platform support for that and then decided it was better not to. Besides, why encourage a fast crap accuracy exp over a fast high accuracy exp.

Comment (pavanky) [inline src/backend/opencl/kernel/bilateral.hpp]: This is image processing. Accuracy really doesn't matter.

Comment (pavanky) [inline src/backend/opencl/kernel/bilateral.hpp]: Anyway removing the blocker for now. Can you also ping me on gitter, i have a couple of questions about this https://gitter.im/arrayfire/arrayfire

**Comment by pavanky** [comment]
@mlloreda can you see what happened to windows here? Merge this if windows is not an issue.

**Comment by mlloreda** [comment]
@pavanky sounds good.

**Comment by mlloreda** [comment]
Turns out this is failing due to changes in devel. Will need re-run PR tests once devel is fixed.

**Comment by mlloreda** [comment]
Okay, the regression in the devel branch has been fixed.

@nevion, can you rebase your PR? We'll make sure to merge right after tests pass.

**Comment by pavanky** [comment]
@mlloreda It doesn't need to be rebased. PR says there are no conflicts. Kick off the tests again.

**Comment by mlloreda** [comment]
build arrayfire windows ci
build arrayfire osx cuda ci

**Pull request merged**